rosidl_generator_c/resource/msg__functions.c.em

```diff
@@ -81,7 +81,10 @@ for field in spec.fields:
         # non-array field
         if field.type.is_primitive_type():
             if field.type.type == 'string':
-                lines.append('rosidl_generator_c__String__init(&msg->%s);' % field.name)
+                lines.append('if (!rosidl_generator_c__String__init(&msg->%s)) %s' % (field.name, '{'))
+                lines.append('    %s__destroy(msg);' % msg_typename)
+                lines.append('  return false;')
```

Comment:
You could e.g. try to construct a partially initialized message manually and then call destroy on it to check that it is cleaned correctly.